docs/transforms.rst

```diff
@@ -0,0 +1,20 @@
+
+.. _transforms-caveats:
+
+Limitations of using JAX transforms inside of networks
+======================================================
+
+Once a Haiku network has been transformed to a pure function using
+:func:`hk.transform() <haiku.transform>`, it's possible to freely combine it with
+any JAX transformations like :func:`jax.jit`, :func:`jax.grad`, and so on.
+It's also possible to use JAX transformations inside of a Haiku network, but
+there can be issues with JAX's tracing mechanism since :func:`hk.transform()
```

Comment:
`hk.transform` isn't the issue - it's `hk.next_rng_key()`, `hk.get_paramter()`, `hk.set_state()`.
What about "since Haiku code that hasn't been `hk.transform()`-ed may have side effects" or something along those lines.